### ISSUE TITLE
Google Drive support

### ISSUE THREAD

**Issue by niksingh710** [Akylas/OSS-DocumentScanner#52]
The app is so awesome and underrated at the same moment.
Purely competent to adobe scan.
just curious if google drive support addition will be possible so that the content can be backed up also.

**Comment by farfromrefug** [comment]
@niksingh710 Thanks for the kind words :D

I did not really plan to add google drive support (because it is google devil thing :D) But well i can think about it.
What are you looking for:
* export pdf to google drive?
* sync app data with google drive (not pdf, app specific format data, like done with webdav) so you dont loose anything if you uninstall/install the app ?

**Comment by niksingh710** [comment]
@farfromrefug 
I am expecting the 2nd pt.
as exporting pdf to google drive can be done easily by exporting to file and then moving to drive or some files app also allows to directly access drive.

i agree with the google's devil thing but ig if rclone is added or in someway that the app stores data to drive in encrypted form to drive (just an optional feature that user can opt for) will make it much better. as imo adobe just have their 2GB free storage where they store all the data and i don't think if they even encrypt it.

**Comment by farfromrefug** [comment]
@niksingh710 i understand. Encrypting would make things even harder though :s. I ll keep that open but wont be happening soon except if more people ask for it

**Comment by Subhashis2007** [comment]
It kinda seems out of context for a scanner app?? since everyone doesn't use Google Drive. And adding such function would make it bloated?? @niksingh710

**Comment by niksingh710** [comment]
> It kinda seems out of context for a scanner app?? since everyone doesn't use Google Drive. And adding such function would make it bloated?? @niksingh710

i agree with you but as adobe provides their own cloud backup it seems reasonable to me if a single option can be provided where doing backup is possible.
as webDAV requires a server 
i am fine with not adding Google Drive, but if rclone is possible to add then also it will be fine.

**Comment by niksingh710** [comment]
Drop box can be a beter alternative as gdrive will introduce google's complex api.

**Comment by farfromrefug** [comment]
@niksingh710 i have looked a bit and i dont think i will implement rclone in my app. It seems totally out of the scope. 
But i found out you could use https://github.com/x0b/rcx to sync a folder from your phone, then set that folder as the default export folder for PDFs.
I have a looked a bit at using something similar to "backup" app data but it does not seem that easy right now. Will test a bit more

**Comment by niksingh710** [comment]
Seems reasonable.
Yeah, I prefer `rcx` but the folder where the data is stored by the `oss-documentscanner` is not exposed to other apps or file manager.

**Comment by farfromrefug** [comment]
yes this is what I started to look at (if you want to sync app data and not PDF).
SAF us a nightmare on android latest androis versions

**Comment by antrmn** [comment]
This feature request could be generalized to all the cloud providers that allow to access storage through a SAF document provider:

- NextCloud 
- Round Sync, which is a fork of RCX that also allows to access any rclone remote over SAF

Google Drive does not allow to select a directory through SAF, but one could use Round Sync for this.

![Screenshot_20241002-194607_Files](https://github.com/user-attachments/assets/ce61a2b2-e9a1-43e5-a73e-50404dd66c5d)

**Comment by farfromrefug** [comment]
@antrmn cant you already do that ? NOw you can sync to a local folder. Theoretically  you should be able to select a folder in nextcloud or roundsync as in your screenshot. Did you try?

**Comment by antrmn** [comment]
Indeed, it is possible to select them, but as soon as I submit a document or tap the "Sync button" I get an error.

This is the error I get when selecting a directory in NextCloud:

![Screenshot_20241002-225203_OSS Document Scanner](https://github.com/user-attachments/assets/3b743749-63e5-4533-8d08-8129e42d0c7f)


This is what I get instead with Round Sync:

![Screenshot_20241002-225458_OSS Document Scanner](https://github.com/user-attachments/assets/ea03b8e2-7b66-4ff1-8327-f0dc3008fa6e)

**Comment by farfromrefug** [comment]
@antrmn sorry To hear that. Will try to reproduce

**Comment by farfromrefug** [comment]
@antrmn dumb question. How did you add Nextcloud as en entry on your drawer menu ?

**Comment by antrmn** [comment]
Just install the Nextcloud app from either F-Droid or Play Store and log in, it should appear

**Comment by farfromrefug** [comment]
@antrmn just released a new version which should fix it!

**Comment by antrmn** [comment]
Can confirm that it works! Thank you :)

**Comment by HikaruIchijyo** [comment]
So what is the result.
What to do 🥲 fr sync?
Can we use nextcloud or gdrive or any other storage?

Not seperate app ofc

**Comment by semi710** [comment]
I do feel that adding support for proprietary or other cloud providers is out of the scope of this project.
However, from the discussion on this thread, it seems that we can now use external cloud sync tools on our devices. These tools can sync a directory from our device, and we can store the `OSS-DocumentScanner` content in that synced directory to enable cross-device synchronization.

Some better tools have already been referenced above by @antrmn at: [https://github.com/Akylas/OSS-DocumentScanner/issues/52#issuecomment-2389257654](https://github.com/Akylas/OSS-DocumentScanner/issues/52#issuecomment-2389257654)
I already use RoundSync on my devices, so this more or less solves my issue as well.

**Comment by semi710** [comment]
> Can we use Nextcloud or Google Drive or any other storage?

Yes.

> Not a separate app, of course.

You need the **Nextcloud** app to sync a directory to Nextcloud. Then, you can configure **OSS-DocumentScanner** to store its content in that synced directory.

**Comment by farfromrefug** [comment]
@niksingh710 @antrmn I am happy to report that google drive and onedrive sync is now implemented in the app!
